src/container/srv_target.c

```diff
@@ -2537,3 +2660,114 @@ ds_cont_ec_timestamp_update(struct ds_cont_child *cont)
 {
 	cont->sc_ec_update_timestamp = d_hlc_get();
 }
+
+void
+ds_cont_child_insert_dtx(struct ds_cont_child *cont_child, struct dtx_leader_handle *new_dlh)
+{
+	struct dtx_leader_handle *dlh = NULL;
+
+	/* Open DTXs should be ordered in the list, and most likely the new inserted DTX
+	 * should be inserted at the head of the list.
+	 */
+	d_list_for_each_entry(dlh, &cont_child->sc_open_dtx_list, dlh_link_list) {
+		if (dlh->dlh_handle.dth_epoch < new_dlh->dlh_handle.dth_epoch) {
+			d_list_add_tail(&new_dlh->dlh_link_list, &dlh->dlh_link_list);
+			return;
+		}
+	}
+
+	d_list_add_tail(&new_dlh->dlh_link_list, &cont_child->sc_open_dtx_list);
+}
+
+static void
+cont_child_update_commit_eph(struct ds_cont_child *cont_child)
+{
+	struct dtx_leader_handle *dlh = NULL;
+
+	if (!cont_child->sc_dtx_commit_eph)
+		return;
+
+	/* First Get the minimum epoch from COS list */
+	*cont_child->sc_dtx_commit_eph = dtx_get_min_cos_eph(cont_child);
+
+	/* Then compare it with the open list, since the open list is naturally sorted,
+	 * so only compare the last one */
+	d_list_for_each_entry_reverse(dlh, &cont_child->sc_open_dtx_list, dlh_link_list) {
```

Comment:
the last one is the highest epoch, seems you should compare with the lowest epoch of the opened DTX right?

Reply:
The last one is the oldest.

Reply:
you are right, thx